sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -1287,4 +1276,26 @@ private[hive] object HiveClientImpl extends Logging {
     }
     hiveConf
   }
+
+  /**
+   * Initialize Hive through Configuration.
+   * first try to use getWithoutRegisterFns to initialize to avoid loading all functions,
+   * if there is no such method, fallback to Hive.get.
+   */
+  def getHive(conf: Configuration): Hive = {
+    conf match {
+      case hiveConf: HiveConf =>
+        try {
+          classOf[Hive].getMethod("getWithoutRegisterFns", classOf[HiveConf])
+            .invoke(null, conf).asInstanceOf[Hive]
+        } catch {
+          // SPARK-37069: not all Hive versions have the above method (e.g., Hive 2.3.9 has it but
+          // 2.3.8 don't), therefore here we fallback when encountering the exception.
+          case _: NoSuchMethodException =>
+            Hive.get(hiveConf)
+        }
+      case _ =>
+        Hive.get(conf, classOf[HiveConf])
```

Comment:
Maybe we can also call `getWithoutRegisterFns` for this path? by first converting the `Configuration` to a `HiveConf`.